### PR TITLE
feat: OR-1389 fork public network

### PR DESCRIPTION
**Pull request by 0x6e616d**
## Description
Fork the public network by using `anvil`

## How to test it
1. Create `.env` file
```sh
export FORK_PUBLIC_NETWORK=true
export L1_RPC='change me'
export FROM_BLOCK_NUMBER=19277372
export L2_NATIVE_TOKEN=0x2be5e8c109e2197D077D13A82dAead6a9b3433C5
export IMPL_SALT='change me'
```
2. Export these envs 
```sh
export $(xargs < .env)
``` 

3. Run
```sh
make devnet-up
```

### CODE REVIEW

**Comment by 0x6e616d** [comment]
> Thank you very much! 🙇🏼‍♂️ Do we need these functions? (anvil_dumpState, convert_anvil_dump) Are you making this for later? If so, I'm curious. Do we use anvil node? (About, anvil_dumpState JSON RPC method)

I am using `anvil` to fork the public network, and `anvil` doesn't support `dump_block` endpoint, instead of that, it has a new endpoint `anvil_dumpState`, which is similar to `dump_block` from geth

Oh I missed my code, I updated

File: bedrock-devnet/devnet/__init__.py [hunk 1]
```diff
@@ -126,18 +137,22 @@ def deploy_contracts(paths):
...
-    # send some ether to the create2 deployer account
-    run_command([
-        'cast', 'send', '--from', account,
-        '--rpc-url', 'http://127.0.0.1:8545',
-        '--unlocked', '--value', '1ether', '0x3fAB184622Dc19b6109349B94811493BF2a45362'
-    ], env={}, cwd=paths.contracts_bedrock_dir)
+    # Proxy exists on the fork public network(used by anvil)
+    # We don't need to deploy the proxy contract
+    # https://book.getfoundry.sh/tutorials/create2-tutorial
+    if not paths.fork_public_network:
```
Comment (rlgns98kr) [inline bedrock-devnet/devnet/__init__.py:143]: I think we have to need some ether to deploy some contracts although it is the fork public network. Is it resolved in the other way?

Comment (0x6e616d) [reply bedrock-devnet/devnet/__init__.py:143]: Yes, anvil provides for us the default accounts for deploying some contracts, and we use `eth_accounts` function to get these accounts, work as the same way with `geth`

Comment (rlgns98kr) [reply bedrock-devnet/devnet/__init__.py:143]: Oh, I'm sorry. I understood it. Thank you for your kindness! 🙇🏼‍♂️

**Review by rlgns98kr — APPROVED** [review]
Thank you very much! 🙇🏼‍♂️

**Pull request merged**